src/main/java/com/example/crm_gym/services/TrainingService.java

```diff
@@ -0,0 +1,124 @@
+package com.example.crm_gym.services;
+
+import com.example.crm_gym.dao.TrainingDAO;
+import com.example.crm_gym.dao.UserDAO;
+import com.example.crm_gym.exception.DaoException;
+import com.example.crm_gym.models.*;
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Slf4j
+@Transactional
+@Service
+public class TrainingService {
+
+    private TrainingDAO trainingDAO;
+    private UserDAO userDAO;
+
+    @Autowired
+    public TrainingService(TrainingDAO trainingDAO, UserDAO userDAO) {
+        this.trainingDAO = trainingDAO;
+        this.userDAO = userDAO;
+    }
+
+    public boolean create(String trainingName, TrainingType trainingType, Date trainingDate, int trainingDuration) {
+        try {
+            Training training = new Training(trainingName, trainingType, trainingDate, trainingDuration);
+            return trainingDAO.save(training);
+        } catch (DaoException e) {
+            log.error("Error creating training", e);
+            return false;
+        }
+    }
+
+    public boolean update(String username, String password, Long id, Training training) {
+        try {
+            if (!userDAO.checkUsernameAndPassword(username, password)) {
```

Comment:
you have repeated code. Read about DRY principle. You can create separate method for that